File: generated_code/cpp/solution_4271_2.cpp
#include <vector>
#include <numeric>

using namespace std;

class Solution {
public:
    vector<int> findDegrees(vector<vector<int>>& matrix) {
        int n = matrix.size();
        vector<int> ans(n, 0);
        
        // The degree of a vertex i in an adjacency matrix is the sum of its row elements.
        // For an undirected graph without self-loops, summing matrix[i][j] gives the count of edges connected to i.
        for (int i = 0; i < n; ++i) {
            ans[i] = std::accumulate(matrix[i].begin(), matrix[i].end(), 0);
        }
        
        return ans;
    }
};

File: generated_code/python/solution_4271_2.py
from typing import List

class Solution:
    def findDegrees(self, matrix: List[List[int]]) -> List[int]:
        """
        Calculates the degree of each vertex in an undirected graph.
        
        Algorithm:
        The degree of a vertex i in an adjacency matrix is simply the sum of 
        the elements in its corresponding row (matrix[i]), since matrix[i][j] = 1 
        indicates an edge and matrix[i][j] = 0 indicates no edge.
        
        Time Complexity: O(n^2) where n is the number of vertices, as we visit each cell once.
        Space Complexity: O(1) auxiliary space (excluding the output array).
        """
        # Sum each row to get the degree of each vertex
        return [sum(row) for row in matrix]